### ISSUE TITLE
display count of messages in sent survey

### ISSUE THREAD

**Issue by crcommons** [ushahidi/platform#2681]
## Requirements
Once API work is complete:
- [ ] display in step 4 "We are processing X number of messages. We have sent Y number and you've gotten Z." after survey is sent.

**Comment by kinstelli** [comment]
I wanted to clarify some of this, since various devs have been discussing it. I think we initially hoped for the information provided to the user to be very simple, but some of us have concerns about how this is potentially misleading/confusing. 

Below are some of my concerns about providing data in the format described above. ( @rowasc , feel free to add/edit this to include your concerns as well)                 
@crcommons @justinscherer, @Erioldoesdesign                          -- tagging you in case you have comments.                                                        

1. Since there's potentially a lag between messages that have been set to 'pending' and messages that have been set to 'sent', maybe it's worth differentiating these for the user.

2. Likewise, it's probably worth differentiating between initial messages sent, as well as subsequent messages sent. (For example, 
"Question 1: 700 sent, 300 pending.
Responses: 312 responses to Question 1
 Question 2: 300 sent, 12 pending. ") 
This data might not be too hard to grab from `targeted_survey_state`, but I'm not certain.

3. I don't think we're currently logging the moment that something has been marked as sent. There is no `sent_datetime` field in `messages`, so we'd have to add it. There is an `updated` field in the `messages` table, and we could infer that this is when it was marked as `sent`, but that seems untrustworthy.
@willdoran  - given what currently happens with dataproviders ( #2674 ) I'm guessing that adding a `sent_datetime` and having it get updated might mean opening up a complicated ticket? 

    In other words, I think including the exact date/time of sent messages might be complicated and confusing -- both to collect and to display to the user -- and maybe it's sufficient to tell the user the current status, as in item 2 above?

**Comment by crcommons** [comment]
I agree this is very confusing. Also given the fact that messages are not all sent at one data and time, it will be difficult to have something like We sent your survey to {{number of recipients}} people on {{data}} at {{time}} UTC, which was in the design. 

I definitely agree we should have a sent and pending. Perhaps a message that says "We're processing your messages" when sent for Q1 is still <1, then the "sent" and "pending" once we've actually started sending them. It also would make sense to break it into sent/pending for each question.

**Comment by justinscherer** [comment]
Good to know that this complexity exists. What does "pending" actually mean in this context? @kinstelli @crcommons

**Comment by willdoran** [comment]
@kinstelli To be useful or meaningful the amount of data we would have to show users would have to be in the vein of what you've marked in point 2.

I would be concerned that the single line in the design would be misleading because it implies a moment in time that will be fleeting at best. We can at the moment do the following sort messages by targeted survey, by question, by status and draw out the numbers you've identified.

Adding a sent timestamp to the messages is straightforward.

The status meaning the following:
- Pending - waiting to be sent out by our system. This happens on a scheduled window every 5-15 mins
- Sent - we've sent it
- Failed - thing didn't go well

It may be very difficult to compress the states down to a single sentence unfortunately without eliding a lot of information and potentially providing in accurate information. We could simplify it to:

"We are processing X number of messages. We have sent Y number and you've gotten Z."

In future iteration, we need some kind of "visualisation" dredged from the depths of 2012 to show what's happening or we could have a table with the states of different Questions
Q1 Sent, Replied Percentage Response rate etc

**Comment by justinscherer** [comment]
> "It may be very difficult to compress the states down to a single sentence unfortunately without eliding a lot of information and potentially providing in accurate information. We could simplify it to:
"We are processing X number of messages. We have sent Y number and you've gotten Z."

The simplified version sounds good to me for now. Let's follow up with users and see what kind of info they need before we build in complex sending SMS analytics visualizations.

**Comment by crcommons** [comment]
Closing in favor of #2562
